tests/e2e/parallel/main_test.go

```diff
@@ -2153,72 +2153,322 @@ func TestScanSettingBinding(t *testing.T) {
+	// suite.
+	if err := f.Client.Create(context.TODO(), origPb, nil); err != nil {
+		t.Fatalf("failed to create ProfileBundle: %s", err)
+	}
+	// This should get cleaned up at the end of the test
+	defer f.Client.Delete(context.TODO(), origPb)
+
+	if err := f.WaitForProfileBundleStatus(pbName, compv1alpha1.DataStreamValid); err != nil {
+		t.Fatalf("failed waiting for the ProfileBundle to become available: %s", err)
+	}
+
+	// Check that the rule exists in the original profile
+	changeTypeRuleName := prefixName(pbName, changeTypeRule)
+	err, found := f.DoesRuleExist(origPb.Namespace, changeTypeRuleName)
+	if err != nil {
+		t.Fatal(err)
+	} else if found != true {
+		t.Fatalf("expected rule %s to exist in namespace %s", changeTypeRuleName, origPb.Namespace)
+	}
+
+	// Check that the rule exists in the original profile
+	unChangedTypeRuleName := prefixName(pbName, unChangedTypeRule)
+	err, found = f.DoesRuleExist(origPb.Namespace, unChangedTypeRuleName)
+	if err != nil {
+		t.Fatal(err)
+	} else if found != true {
+		t.Fatalf("expected rule %s to exist in namespace %s", unChangedTypeRuleName, origPb.Namespace)
+	}
+
+	tpMix := &compv1alpha1.TailoredProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tpMixName,
 			Namespace: f.OperatorNamespace,
+			Annotations: map[string]string{
+				compv1alpha1.PruneOutdatedReferencesAnnotationKey: "",
+			},
 		},
 		Spec: compv1alpha1.TailoredProfileSpec{
-			Title:       "TestCisForE2EPool",
-			Description: "TestCisForE2EPool",
-			Extends:     "ocp4-cis",
-			SetValues: []compv1alpha1.VariableValueSpec{
+			Title:       "TestForManyRules",
+			Description: "TestForManyRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
 				{
-					Name:      "ocp4-var-role-master",
-					Rationale: "targets the e2e pool",
-					Value:     "e2e",
+					Name:      changeTypeRuleName,
+					Rationale: "this rule should be removed from the profile",
 				},
 				{
-					Name:      "ocp4-var-role-worker",
-					Rationale: "targets the e2e pool",
-					Value:     "e2e",
+					Name:      unChangedTypeRuleName,
+					Rationale: "this rule should not be removed from the profile",
 				},
 			},
 		},
 	}
 
-	createTPErr := f.Client.Create(context.TODO(), tp, nil)
-	if createTPErr != nil {
-		t.Fatal(createTPErr)
+	tpSingle := &compv1alpha1.TailoredProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tpSingleName,
+			Namespace: f.OperatorNamespace,
+			Annotations: map[string]string{
+				compv1alpha1.PruneOutdatedReferencesAnnotationKey: "",
+			},
+		},
+		Spec: compv1alpha1.TailoredProfileSpec{
+			Title:       "TestForManyRules",
+			Description: "TestForManyRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      changeTypeRuleName,
+					Rationale: "this rule should be removed from the profile",
+				},
+			},
+		},
 	}
-	defer f.Client.Delete(context.TODO(), tp)
 
-	scanSettingBinding := compv1alpha1.ScanSettingBinding{
+	tpSingleNoPrune := &compv1alpha1.TailoredProfile{
```

Comment:
Nit: `tpSingleNoPrune` should actually be called  `tpMixNoPrune`, as it has two rules in `EnabledRules`

```suggestion
	tpMixNoPrune := &compv1alpha1.TailoredProfile{
```